src/core/mac/mac.hpp

```diff
@@ -576,7 +576,19 @@ class Mac : public InstanceLocator, private NonCopyable
      * @param[in]  aEnable The requested State for the MAC layer. true - Start, false - Stop.
      *
      */
-    void SetEnabled(bool aEnable) { mEnabled = aEnable; }
+    void SetEnabled(bool aEnable)
+    {
+        mEnabled = aEnable;
+
+        if (aEnable)
+        {
+            mLinks.Enable();
+        }
+        else
+        {
+            mLinks.Disable();
+        }
```

Comment:
Should we move the implementation to `cpp` (now that its larger than simple setter)?